Fix validator silent errors and ensure process exit on validation error

pjlee11 opened the pull request:
Resolves #853 #854 

1) Fixes a syntax error in the fixture file `data/test/news/articles/c0000000005o.json` (Issue 854)
2) Adds logic to exit the process which kills the `npm run dataValidate` job upon seeing a failure
3) Move the dataValidate command before the unit tests to ensure the unit tests do not silently error and timeout due to the newly added `process.exit()`

- [x] Tests added for new features
- [ ] Test engineer approval

Commit:
fix syntax error in fixture data

Commit:
exit process on an error and log failing fixture filepath

Commit:
Move the dataValidate command before the unit tests

This is required because now the validator has logic to process.exit() the unit tests will hang and not timeout/error if the process is exited due to an validation error. By running the validator first this ensure no timeout will occur in the unit tests.

pjlee11 commented:
On hold awaiting the refinement of #853 and #854

bcmn reviewed (dismissed):
Works great 👍 

Edit: Ooops, I didn't see this was on hold. 🤐

pjlee11 commented:
With the highlighted data removed the command `npm run dataValidate` now exits the process.
<img width="1150" alt="datavalidate failing correctly" src="https://user-images.githubusercontent.com/7791726/47579300-75f7a400-d943-11e8-9f07-c430e05c5e25.png">

With the highlighted data removed the command `npm run test` now runs the data validation prior to the unit tests to ensure no timeout on a process.exit while the test suite running.
<img width="1154" alt="validate failing before unit tests" src="https://user-images.githubusercontent.com/7791726/47579301-75f7a400-d943-11e8-99ed-750e0866cf33.png">

Commit:
Woohoo I got the test to pass

- adds an invalid scenario for the test to fail validation and therefore exit the process
- stubs the process.exit method so that it doesn't timeout with a real process exit

Commit:
Merge branch 'latest' of github.com:BBC-News/simorgh into ensure-validator-exits-process-on-error

jamesbrumpton committed:
Merge branch 'latest' into ensure-validator-exits-process-on-error

Commit:
Merge branch 'latest' into ensure-validator-exits-process-on-error
